### PR TITLE
Add shaman permission input

### PR DESCRIPTION
**Pull request by alexkeating**
## GitHub Issue

#574 

## Changes

Add a shaman permission input to core app

### CODE REVIEW

File: libs/common-utilities/src/constants/contracts.ts [hunk 1]
```diff
@@ -36,3 +36,38 @@ export const CONTRACTS: ContractAddressList = {
     '0x5': '0x000000000000cd17345801aa8147b8d3950260ff',
   },
 };
+
```
Comment (skuhlmann) [inline libs/common-utilities/src/constants/contracts.ts:39]: see unused import on this one.

File: apps/core-app/src/components/ShamanInput.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Select } from '@daohaus/ui';
```
Comment (skuhlmann) [inline apps/core-app/src/components/ShamanInput.tsx:1]: Let's move this into the UI library as a moecule

File: apps/core-app/src/components/ShamanInput.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+import { SHAMAN_PERMISSIONS } from '@daohaus/common-utilities';
+
+export const ShamanInput = () => {
+  const options = Object.values(SHAMAN_PERMISSIONS).map((permission) => ({
+    name: permission.displayName,
+    value: permission.id,
+  }));
+  return <Select id="shaman-input" options={options} />;
```
Comment (skuhlmann) [inline apps/core-app/src/components/ShamanInput.tsx:9]: Use WrappedSelect here

File: libs/ui/src/types/formAndField.ts [hunk 4]
```diff
@@ -36,14 +36,17 @@ export type PrimitiveSizable = {
 };
 
 export type PrimitiveAddons = {
-  rightAddon?: React.ReactNode
-}
+  rightAddon?: React.ReactNode;
+};
```
Comment (skuhlmann) [inline libs/ui/src/types/formAndField.ts:40]: Thinking we might not want to extend right add on to all fields?

**Review by skuhlmann — CHANGES_REQUESTED** [review]

File: libs/ui/src/types/formAndField.ts (hunk 4 above)
Comment (alexkeating) [reply libs/ui/src/types/formAndField.ts:40]: This is only a formatting change change, there should be changed from what already exists

File: libs/common-utilities/src/constants/contracts.ts (hunk 1 above)
Comment (alexkeating) [reply libs/common-utilities/src/constants/contracts.ts:39]: Are you referring to the first line of the file? I can remove that

File: apps/core-app/src/components/ShamanInput.tsx (hunk 3 above)
Comment (alexkeating) [reply apps/core-app/src/components/ShamanInput.tsx:9]: Will update

File: apps/core-app/src/components/ShamanInput.tsx (hunk 2 above)
Comment (alexkeating) [reply apps/core-app/src/components/ShamanInput.tsx:1]: Will do

File: libs/ui/src/types/formAndField.ts (hunk 4 above)
Comment (skuhlmann) [reply libs/ui/src/types/formAndField.ts:40]: > This is only a formatting change change, there should be changed from what already exists

ah whoops. got it.

Comment (jordanlesich) [reply libs/ui/src/types/formAndField.ts:40]: Let's figure out where that got put in and address it.

Comment (skuhlmann) [reply libs/ui/src/types/formAndField.ts:40]: > Let's figure out where that got put in and address it.

@jordanlesich 

It was added with this one that says to add it to fieldwrapper:
https://github.com/HausDAO/daohaus-monorepo/issues/568

Where would you rather see it? I'll open a new ticket to move it.

Comment (jordanlesich) [reply libs/ui/src/types/formAndField.ts:40]: Oh, it's just for the wrapper. I thought it was for the input. My bad. This is exactly what we want.

File: libs/common-utilities/src/constants/contracts.ts (hunk 1 above)
Comment (alexkeating) [reply libs/common-utilities/src/constants/contracts.ts:38]: I removed the unused import above

**Comment by alexkeating** [comment]
Hopefully should be good now! Thanks for the review

**Review by skuhlmann — APPROVED** [review]

**Pull request merged**